code/tailer/datastripe.q

```diff
@@ -27,33 +27,55 @@ modaccess:{[accesstab]};
     modaccess[.ds.access];
 
     // call the savedown function
-    .ds.savealltablesoverperiod[.ds.td;nextp;lasttime];
+    .ds.savealltablesoverperiod[.ds.td;lasttime];
     .lg.o[`reload;"Kept ",string[.ds.periodstokeep]," period",$[.ds.periodstokeep>1;"s";""]," of data from : ",", " sv string[t]];
     
     // update the access table on disk
-    atab:get ` sv(.ds.td;.proc.procname;`access);
+    accesspath: ` sv(.ds.td;.proc.procname;`$ string .wdb.currentpartition;`access);
+    atab:get accesspath;
     atab,:() xkey .ds.access;
-    (` sv(.ds.td;.proc.procname;`access)) set atab;
+    accesspath set atab;
 
     // update the access table in the gateway
     handles:(.servers.getservers[`proctype;`gateway;()!();1b;1b])[`w];
     .ds.updategw each handles;
 
     };
 
+.wdb.datastripeendofday:{[pt;processdata]
+    //save all tables
+    .ds.savetables[.ds.td;] each .wdb.tablelist[];
+    //clear tables
+    @[`.;;0#] each .wdb.tablelist[];
+    //move to next partition
+    .wdb.currentpartition:pt+1;
+    //create accesspath
+    accesspath: ` sv(.ds.td;.proc.procname;`$ string .wdb.currentpartition;`access);
+    //make access for next partition
+    t:tables[`.] except .wdb.ignorelist;
```

Comment:
Is there a reason you don't use .wdb.tablelist[] here? Think it is the same, unless something happens at EOD that I have forgotten